File: Client/src/common/models/service.ts
import { Profile } from './core/profile';
import { Company } from './company';
import { Product } from './product';
import { Software } from './software';

export class Service extends Profile<Service>{
  public companyId?: number;
  public productId?: number;
  public softwareId?: number;
  public company?: Company;
  public product?: Product;
  public software?: Software;
}

File: Server/Resources/Models/Service.cs
using Server.Resources.Models.Core;

namespace Server.Resources.Models
{
    public class Service : Profile<Service>
    {
        public int? CompanyId { get; set; }
        public int? ProductId { get; set; }
        public int? SoftwareId { get; set; }
        public virtual Company Company { get; set; }
        public virtual Product Product { get; set; }
        public virtual Software Software { get; set; }
    }
}